ruma-events/src/room/message/content_serde.rs

```diff
@@ -1,24 +1,125 @@
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            "m.location" => Self {
+                msgtype: MessageType::Location(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            "m.notice" => Self {
+                msgtype: MessageType::Notice(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            "m.server_notice" => Self {
+                msgtype: MessageType::ServerNotice(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            "m.text" => Self {
+                msgtype: MessageType::Text(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            "m.video" => Self {
+                msgtype: MessageType::Video(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            #[cfg(feature = "unstable-pre-spec")]
+            "m.key.verification.request" => Self {
+                msgtype: MessageType::VerificationRequest(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+            _ => Self {
+                msgtype: MessageType::_Custom(from_raw_json_value(&json)?),
+                relates_to: relates_to.map(|json| from_raw_json_value(&json)).transpose()?,
+                #[cfg(feature = "unstable-pre-spec")]
+                new_content: new_content.map(|json| from_raw_json_value(&json)).transpose()?,
+            },
+        })
```

Comment:
It might be that this will run into the same deserialization error you got with the derived impl, we'll see. But I'd expect this, if we need this impl at all:

```suggestion
        Ok(Self {
            msgtype: from_raw_json_value(&json)?,
            relates_to,
            #[cfg(feature = "unstable-pre-spec")]
            new_content,
        })
```

Reply:
It works YAY! MUCH cleaner.